src/ucp/core/ucp_mm.c

```diff
@@ -886,15 +887,16 @@ ucp_memh_find_slow(ucp_context_h context, void *address, size_t length,
         ucs_ptr_check_align(ucp_memh_address(memh), ucp_memh_length(memh),
                             align);
 
-        if (ucs_test_all_flags(memh->uct_flags,
-                               uct_flags & UCP_MM_UCT_ACCESS_MASK)) {
+        if (ucs_test_all_flags(memh->uct_flags, access_flags)) {
             *memh_p = memh;
             return UCS_OK;
         }
 
         ucs_trace("memh %p uct_flags mismatch, expected 0x%x got 0x%x", memh,
-                  uct_flags & UCP_MM_UCT_ACCESS_MASK,
-                  memh->uct_flags & UCP_MM_UCT_ACCESS_MASK);
+                  access_flags, memh->uct_flags & UCP_MM_UCT_ACCESS_MASK);
+
+        /* Augment permissions to avoid back and forth invalidation */
+        uct_flags |= (memh->uct_flags & UCP_MM_UCT_ACCESS_MASK);
```

Comment:
It looks like the macro can be changed to:
```C
#define UCP_MM_UCT_ACCESS_MASK(_flags) (_flags & UCT_MD_MEM_ACCESS_ALL)
```

Reply:
fixed